trace-server/org.eclipse.tracecompass.incubator.trace.server.jersey.rest.core/src/org/eclipse/tracecompass/incubator/internal/trace/server/jersey/rest/core/services/ExperimentManagerService.java

```diff
@@ -349,9 +350,15 @@ public Response postExperiment(@RequestBody(content = {
 
         // Instantiate the experiment and return it
         ITmfTrace[] traces = Lists.transform(traceUUIDs, uuid -> TraceManagerService.createTraceInstance(uuid)).toArray(new ITmfTrace[0]);
+        Map<UUID, ITmfTrace> uuidToTraceInstance = new HashMap<>();
+        for (UUID uuid : traceUUIDs) {
+            ITmfTrace trace = TraceManagerService.createTraceInstance(uuid);
```

Comment:
I think this can all be simplified:
At line 352 we have the ordered list of traceUUIDs that can later be obtained in Experiment.from(TmfExperiment, UUID) from getTraceUUIDs(expUUID).
The ordered list of created trace instances is fed to the experiment in initExperiment(). This list can also be later obtained in Experiment.from(TmfExperiment, UUID) from TmfExperiment.getTraces().
So if you iterate by index in both lists at the same time, you can access the corresponding values needed for Trace.from(ITmfTrace, UUID).

Reply:
Done.

Reply:
Perhaps to address the first comment, build the uuidToTracesInstance map from traceUUIDs as the traces are being created, then after create the trace array from the map values. Would need to make sure to use a LinkedHashMap to preserve order.